src/lib/storage/index/partial_hash/partial_hash_index.cpp

```diff
@@ -0,0 +1,80 @@
+#include "partial_hash_index.hpp"
+
+#include "storage/segment_iterate.hpp"
+
+namespace hyrise {
+
+PartialHashIndex::PartialHashIndex(const std::vector<std::pair<ChunkID, std::shared_ptr<Chunk>>>& chunks_to_index,
+                                   const ColumnID column_id)
+    : _column_id{column_id} {
+  Assert(!chunks_to_index.empty(), "PartialHashIndex requires chunks_to_index not to be empty.");
+  resolve_data_type(chunks_to_index.front().second->get_segment(column_id)->data_type(),
+                    [&](const auto column_data_type) {
+                      using ColumnDataType = typename decltype(column_data_type)::type;
+                      _impl = std::make_unique<PartialHashIndexImpl<ColumnDataType>>(chunks_to_index, _column_id);
+                    });
+}
+
+size_t PartialHashIndex::insert_entries(
+    const std::vector<std::pair<ChunkID, std::shared_ptr<Chunk>>>& chunks_to_index) {
+  // Prevents multiple threads from indexing the same chunk concurrently.
+  auto lock = std::lock_guard<std::shared_mutex>{_data_access_mutex};
```

Comment:
Pick here: I thin use can use `const auto&` for your locks.

Reply:
```
/hyrise/src/lib/storage/index/partial_hash/partial_hash_index.cpp:21:15: error: unused variable 'lock' [-Werror,-Wunused-variable]
  const auto& lock = std::lock_guard<std::shared_mutex>{_data_access_mutex};
```

The following works, but is quite verbose:
```
[[maybe_unused]] const auto& lock = std::lock_guard<std::shared_mutex>{_data_access_mutex};
```

Instead, for the sake of readability, I would prefer
```
const auto lock = std::lock_guard<std::shared_mutex>{_data_access_mutex};
```

Reply:
Interesting :D
Nvm then